pkg/cache/application_test.go

```diff
@@ -1282,11 +1288,33 @@ func TestApplication_onReservationStateChange(t *testing.T) {
 	task3.placeholder = true
 	app.onReservationStateChange()
 	assertAppState(t, app, ApplicationStates().Accepted, 1*time.Second)
-
+	
 	// app moves to running
 	task3.setTaskGroupName("test-group-2")
 	app.onReservationStateChange()
 	assertAppState(t, app, ApplicationStates().Running, 1*time.Second)
+
+	message := "Placeholder Allocated"
+	reason := "Placeholder Allocated"
+	counter := 0
+	// check that the event has been published
+	err := utils.WaitForCondition(func() bool {
+		for {
+			select {
+			case event := <-recorder.Events:
+				print(event)
+				if strings.Contains(event, reason) && strings.Contains(event, message) {
+					counter++
+					if counter == 4 {
+						return true
+					}
+				}
+			default:
+				return false
+			}
+		}
+	}, 5*time.Millisecond, 20*time.Millisecond)
```

Comment:
20ms seems to be a very low value, this can be a problem on slower machines. Use a safer value like `time.Second`.